### ISSUE TITLE
Epic: Release 1.0.0

### ISSUE THREAD

**Issue by hqdncw** [hqdncw/repo_checksums#1]
### Big Picture

<details>

  <summary>test assignment</summary>
  
  ```sh
  Напишите скрипт, асинхронно, в 3 одновременных задачи, скачивающий содержимое HEAD репозитория https://gitea.radium.group/radium/project-configuration во временную папку.
  После выполнения всех асинхронных задач скрипт должен посчитать sha256 хэши от каждого файла.
  Код должен проходить без замечаний проверку линтером wemake-python-styleguide. Конфигурация nitpick - https://gitea.radium.group/radium/project-configuration
  Обязательно 100% покрытие тестами

  Прошу ответить на следующие вопросы:
  Как вы реализовали асинхронное выполнение задач в вашем скрипте?
  Какие библиотеки использовались для скачивания содержимого репозитория и для каких целей?
  Какие проблемы асинхронности вы сталкивались при выполнении задания и как их решали?
  Как вы организовали скачивание файлов во временную папку?
  Какие основные требования wemake-python-styleguide вы находите наиболее важными для поддержания качества кода?
  Как вы настраивали свой проект для соответствия конфигурации nitpick, указанной в задании? Были ли трудности при настройке?
  Какие инструменты использовали для измерения 100% покрытия тестами?
  Какие типы тестов вы написали для проверки функциональности вашего скрипта? (Например, модульные тесты, интеграционные тесты)
  Как вы тестировали асинхронный код? Использовали ли вы моки (mocks) или стабы (stubs) для тестирования асинхронных операций?
  ```
  
</details>

Build a command-line tool to hash files.

### Open Questions

- [ ] #5
- [ ] #4

### Blockers / Dependencies

- [ ] #5

#### Essential Features (p1)

- [ ] Calculate SHA256 hashes using the CPU
- [ ] Output the results to the terminal
- [ ] #3 
- [ ] Integrate code quality checks using [wemake-python-styleguide](https://github.com/wemake-services/wemake-python-styleguide)

#### Nice-to-Haves (p2)

- [ ] Consider adding an option to calculate SHA256 hashes in parallel
- [ ] Consider adding an option to offload SHA256 hash calculations to the GPU
- [ ] Run benchmarks on a variety of datasets with different sizes, such as the Chromium repo (with 400K working tree files and a 36 GiB repo size)
- [ ] Hash files as they're fetched, rather than waiting for the entire Git fetch to complete
- [ ] Support different hashing algorithms (e.g., SHA1)
- [ ] Let users add custom headers to file contents before building a hash
- [ ] Support using Git-generated hashes directly, bypassing local hash calculation

**Comment by hqdncw** [comment]
> Can we just use the existing hash that Git generates for each file, or do we need to create our own?

By default, Git uses SHA1, but SHA256 can be opted for when creating a repository with the [--object-format=sha256](https://git-scm.com/docs/git-init#Documentation/git-init.txt-code--object-formatcodeemltformatgtem) flag.

The format that Git uses for its hashes is straightforward. It consists of the object type (blob), the file size in bytes, a null byte to separate the headers from the contents, and then the file contents itself.

**For example:**

Running `git hash-object` on the string `test` yields a hash like this:

```bash
$ printf "test" | git hash-object --stdin
aa19560d465e7d43915547490a1f6b73eb55702e3d12cb82fb577df60bad4928
```

Manually constructing the same hash using sha256sum produces the same result:

```bash
$ printf "blob 4\x00test" | sha256sum
aa19560d465e7d43915547490a1f6b73eb55702e3d12cb82fb577df60bad4928  -
```

See also: https://git-scm.com/book/en/v2/Git-Internals-Git-Objects
